metron-platform/metron-parsing/metron-parsers/src/test/java/org/apache/metron/parsers/leef/LEEFParserTest.java

```diff
@@ -0,0 +1,252 @@
+		JSONObject obj = parse.get(0);
+		Assert.assertNotNull(obj);
+		Assert.assertEquals("3412364560000000008", obj.get("fileId"));
+		Assert.assertEquals("Mozilla/5.0 (Windows NT 6.1; WOW64; rv:38.0) Gecko/20100101 Firefox/38.0", obj.get("requestClientApplication"));
+		Assert.assertTrue(obj.containsKey("longitude"));
+		Assert.assertFalse(obj.containsKey("cs8"));
+		Assert.assertFalse(obj.containsKey("cs8Label"));
+	}
+
+	@Test
+	public void testLEEFParserSample() throws Exception {
+		runTest("sample", Resources.readLines(Resources.getResource(getClass(), "sample.leef"), UTF_8),
+				Resources.toString(Resources.getResource(getClass(), "sample.schema"), UTF_8));
+	}
+
+	private void runTest(String name, List<String> lines, String schema) throws Exception {
+		runTest(name, lines, schema, "");
+	}
+
+	private void runTest(String name, List<String> lines, String schema, String targetJson) throws Exception {
+		for (String inputString : lines) {
+			JSONObject parsed = parse(inputString).get(0);
+			Assert.assertNotNull(parsed);
+			Assert.assertNotNull(parsed.get("timestamp"));
+			Assert.assertTrue((long) parsed.get("timestamp") > 0);
+
+			JSONParser parser = new JSONParser();
+
+			Map<?, ?> json = null;
+			try {
+				json = (Map<?, ?>) parser.parse(parsed.toJSONString());
+				Assert.assertEquals(true, validateJsonData(schema, json.toString()));
+			} catch (ParseException e) {
+				e.printStackTrace();
+			}
+
+			// test against an explicit json example
+			if (!targetJson.isEmpty()) {
+
+			}
+		}
+	}
+
+	private void assertSimpleSample(List<JSONObject> parse) {
+		JSONObject obj = parse.get(0);
+		Assert.assertNotNull(obj);
+		Assert.assertTrue(obj.containsKey("ip_src_addr"));
```

Comment:
All common fields should use the keys defined in the Constants file https://github.com/apache/metron/blob/master/metron-platform/metron-common/src/main/java/org/apache/metron/common/Constants.java#L51